source/geometries/Cigar.cc

```diff
@@ -0,0 +1,512 @@
+    sipm->SetVisibility(true);
+    sipm->SetSensorDepth(1);
+    sipm->SetTimeBinning(50 * ns);
+    sipm->SetWindowRefractiveIndex(opticalprops::OptCoupler()->GetProperty("RINDEX"));
+    sipm->Construct();
+    G4LogicalVolume* sipm_logic = sipm->GetLogicalVolume();
+    // // ALUMINIZED ENDCAP
+
+    G4Material *fiber_end_mat = G4NistManager::Instance()->FindOrBuildMaterial("G4_Al");
+
+    G4double fiber_end_z = 350 * nm;
+
+    G4Tubs *fiber_end_solid_vol =
+        new G4Tubs("fiber_end", 0, fiber_diameter_ / 2, fiber_end_z, 0, 2 * M_PI);
+
+    G4LogicalVolume *fiber_end_logic_vol =
+        new G4LogicalVolume(fiber_end_solid_vol, fiber_end_mat, "FIBER_END");
+    G4OpticalSurface *opsur_al =
+        new G4OpticalSurface("AL_OPSURF", glisur, ground, dielectric_metal);
+    opsur_al->SetPolish(0.75);
+    opsur_al->SetMaterialPropertiesTable(opticalprops::PolishedAl());
+
+    new G4LogicalSkinSurface("AL_OPSURF", fiber_end_logic_vol, opsur_al);
+
+    // // PLACEMENT /////////////////////////////////////////////
+    G4RotationMatrix *rot_y = new G4RotationMatrix();
+    G4RotationMatrix *rot_z = new G4RotationMatrix();
+    rot_y->rotateY(180 * deg);
+    rot_z->rotateZ(90 * deg);
+    new G4PVPlacement(G4Transform3D(*rot_y, G4ThreeVector(cigar_width_ / 2 + fiber_diameter_ / 2+extra_width/2+panel_width/2+fiber_diameter_/4, 0, cigar_length_ / 2 + photosensor_thickness / 2 + 7 * cm)),
+                      sipm_logic, "SIPM1", world_logic_vol,
+                      false, 1, false);
+
+    new G4PVPlacement(G4Transform3D(*rot_y, G4ThreeVector(- cigar_width_ / 2 - fiber_diameter_ / 2-(extra_width/2+panel_width/2+fiber_diameter_/4), 0, cigar_length_ / 2 + photosensor_thickness / 2 + 7 * cm)),
+                      sipm_logic, "SIPM2", world_logic_vol,
+                      false, 2, false);
+    new G4PVPlacement(G4Transform3D(*rot_z, G4ThreeVector(0, - cigar_width_ / 2 - fiber_diameter_ / 2-(extra_width/2+panel_width/2+fiber_diameter_/4), cigar_length_ / 2 + photosensor_thickness / 2 + 7 * cm)),
+                      sipm_logic, "SIPM3", world_logic_vol,
+                      false, 3, false);
+    new G4PVPlacement(G4Transform3D(*rot_z, G4ThreeVector(0, cigar_width_ / 2 + fiber_diameter_ / 2+extra_width/2+panel_width/2+fiber_diameter_/4, cigar_length_ / 2 + photosensor_thickness / 2 + 7 * cm)),
+                      sipm_logic, "SIPM4", world_logic_vol,
+                      false, 4, false);
+
+
+
+    // Teflon special closing panel
+    G4Box* teflon_closing_panel =
+      new G4Box("TEFLON_PANEL_FRONT_TEMP", cigar_width_ / 2 + extra_width+panel_width, cigar_width_ / 2 + extra_width+panel_width, panel_width / 2);
+    // G4LogicalVolume* teflon_logic_front =
+    //   new G4LogicalVolume(teflon_closing_panel, teflon, "TEFLON");
+    // teflon_logic_front->SetVisAttributes(nexus::White());
+    G4VSolid* temp_solid = teflon_closing_panel;
+    G4double fiber_radius = fiber_diameter_/1.1;
```

Comment:
The name of this variable is somewhat misleading: a radius should be a diameter/2. In line 328 it becomes evident that it is not a radius, so change the name to a more meaningful one.